File: C/sort/insertion.c
#include "insertion.h"
#include <locale.h>
int main(int argc, char const *argv[])
{
    setlocale(LC_ALL, "zh_CN.UTF-8"); // 设置为简体中文UTF-8编码
    const int size = 20;
    int *arr = randArr(size, 1, 1000);
    printArr(arr, 20, "排序前:");
    insertionSort(arr, 20);
    printArr(arr, 20, "排序后:");

    /* code */
    return 0;
}

File: dart/src/sort/insertion.dart
// 插入排序升序
void insertionSort(List<int> list) {
  for (int i = 1; i < list.length; i++) {
    int key = list[i];
    int j = i - 1;
    while (j >= 0 && list[j] > key) {
      list[j + 1] = list[j];
      j = j - 1;
    }
    list[j + 1] = key;
  }
}

// 插入排序降序
void insertionSort_Desc(List<int> list) {
  for (int i = 1; i < list.length; i++) {
    int key = list[i];
    int j = i - 1;
    while (j >= 0 && list[j] < key) {
      list[j + 1] = list[j];
      j = j - 1;
    }
    list[j + 1] = key;
  }
}
